core/templates/pages/story-editor-page/editor-tab/story-node-editor.component.spec.ts

```diff
@@ -0,0 +1,634 @@
+    spyOnProperty(storyEditorStateService, 'onRecalculateAvailableNodes')
+      .and.returnValue(mockEventEmitterLast);
+
+    component.nodeId = 'node1';
+    component.storyNodeIds = ['node1', 'node_2', 'node_3', 'wroking'];
+    component.destinationNodeIds = ['node_2'];
+    component.ngOnInit();
+  });
+
+  afterEach(() => {
+    component.ngOnDestroy();
+  });
+
+  it('should init the controller', () => {
+    component.viewNodeEditor('node_1');
+    expect(component.chapterPreviewCardIsShown).toEqual(false);
+    expect(component.mainChapterCardIsShown).toEqual(true);
+    expect(component.explorationInputButtonsAreShown).toEqual(false);
+  });
+
+  it('should return skill editor URL', () => {
+    expect(component.getSkillEditorUrl('skill_1')).toEqual(
+      '/skill_editor/skill_1');
+  });
+
+  it('should fetch the descriptions for prerequisite skills', fakeAsync(() => {
+    component.prerequisiteSkillIds = ['1', '2', '3'];
+
+    component.getPrerequisiteSkillsDescription();
+    tick();
+
+    expect(component.skillIdToSummaryMap).toEqual(
+      { 1: 'test', 2: 'test2', 3: 'test3' }
+    );
+  }));
+
+  it('should call Alerts Service if getting skill desc. fails', fakeAsync(
+    () => {
+      component.prerequisiteSkillIds = ['2'];
+      let alertsSpy = spyOn(alertsService, 'addWarning').and.callThrough();
+
+      component.getPrerequisiteSkillsDescription();
+      tick();
+
+      expect(alertsSpy).toHaveBeenCalled();
+    }));
+
+  it('should check if exploration can be saved', () => {
+    component.checkCanSaveExpId();
+    expect(component.expIdCanBeSaved).toEqual(true);
+  });
+
+  it('should call StoryUpdate service remove prerequisite skill id',
+    () => {
+      let skillSpy = spyOn(
+        storyUpdateService, 'removePrerequisiteSkillIdFromNode');
+      component.removePrerequisiteSkillId('skill_3');
+      expect(skillSpy).toHaveBeenCalled();
+    });
+
+  it('should call StoryUpdate service remove acquired skill id', () => {
+    let skillSpy = spyOn(storyUpdateService, 'removeAcquiredSkillIdFromNode');
+    component.removeAcquiredSkillId('node_1');
+    expect(skillSpy).toHaveBeenCalled();
+  });
+
+  it('should toggle chapter preview card', () => {
+    component.chapterPreviewCardIsShown = false;
+    component.togglePreview();
+    component.chapterPreviewCardIsShown = true;
+
+    component.togglePreview();
+    component.chapterPreviewCardIsShown = false;
+  });
+
+  it('should toggle prereq skill list', () => {
+    component.prerequisiteSkillIsShown = true;
+    component.togglePrerequisiteSkillsList();
+    component.prerequisiteSkillIsShown = false;
+  });
+
+  it('should call StoryUpdate service to set story thumbnail filename',
+    () => {
+      let storySpy = spyOn(storyUpdateService, 'setStoryNodeThumbnailFilename');
+      component.updateThumbnailFilename('new_file.png');
+      expect(storySpy).toHaveBeenCalled();
```

Comment:
```suggestion
      let storySpy = spyOn(storyUpdateService, 'setStoryNodeThumbnailFilename');
 
      component.updateThumbnailFilename('new_file.png');
 
      expect(storySpy).toHaveBeenCalled();
```

Please do the same for the rest of the tests. We usually split a test into pre-check, setup, action and post-check by adding new lines between them.

Reply:
done